graphql_compiler/compiler/emit_sql.py

```diff
@@ -205,23 +252,55 @@ def _set_group_by(self, group_by):
         """Set output columns for the fold object."""
         self._group_by = group_by
 
-    def _construct_fold_joins(self, edge, from_alias, to_alias):
+    def _construct_fold_joins(self):
         """Use the edge descriptors to create the join clause between the tables in the fold."""
-        join_clause = sqlalchemy.join(
-            from_alias,
-            to_alias,
-            onclause=(from_alias.c[edge.from_column] == to_alias.c[edge.to_column])
-        )
+        _, from_alias, _ = self.join_info[-1]
+        join_clause = from_alias
+        if self._is_mssql2014:
+            # The final join predicate (from the vertex preceding the output to the output vertex)
+            # is moved into the WHERE clause of the SELECT ... FOR XML PATH('')
+            self.join_info.pop()
+        while len(self.join_info) > 0:
+            # more recent joins are at the end of the list
+            edge, from_alias, to_alias = self.join_info.pop()
+            join_clause = sqlalchemy.join(
+                join_clause,
+                to_alias,
+                onclause=(from_alias.c[edge.from_column] == to_alias.c[edge.to_column])
+            )
         return join_clause
 
     def _construct_fold_subquery(self, subquery_from_clause):
         """Combine all parts of the fold object to produce the complete fold subquery."""
-        return sqlalchemy.select(
+        select_stmt = sqlalchemy.select(
```

Comment:
don't use abbreviations since it makes it harder for the user to easily understand what you mean.